python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -1227,25 +1236,18 @@ def get_asset_backfill_iteration_materialized_subset(
 
 
 def _get_failed_and_downstream_asset_graph_subset(
-    backfill_id: str,
     asset_backfill_data: AssetBackfillData,
     asset_graph_view: AssetGraphView,
-    materialized_subset: AssetGraphSubset,
     failed_asset_graph_subset: AssetGraphSubset,
 ) -> AssetGraphSubset:
-    failed_and_downstream_subset = bfs_filter_asset_graph_view(
-        asset_graph_view,
-        lambda candidate_asset_graph_subset, _: (
-            AssetGraphViewBfsFilterConditionResult(
-                passed_asset_graph_subset=candidate_asset_graph_subset,
-                excluded_asset_graph_subsets_and_reasons=[],
-            )
-        ),
-        initial_asset_graph_subset=failed_asset_graph_subset,
-        include_full_execution_set=False,
-    )[0]
+    failed_and_downstream_subset = asset_graph_view.compute_downstream_subsets(
+        list(asset_graph_view.iterate_asset_subsets(failed_asset_graph_subset))
+    )
 
-    return failed_and_downstream_subset & asset_backfill_data.target_subset
+    return (
+        AssetGraphSubset.from_entity_subsets(failed_and_downstream_subset)
+        & asset_backfill_data.target_subset
```

Comment:
As per comments on the above Pr in the stack - i am wondering if we should keep the aspect in compute_downstream_subsets where the result is intersected with a subset at each stage. Given that the target subset will often be a very small subset of the asset graph and that is all that we care about in this context

Reply:
resolved upstack